### ISSUE TITLE
make parameters in NSolver constant

### ISSUE THREAD

**Issue by QiaoLei-88** [QiaoLei-88/NSolver#98]

**Comment by QiaoLei-88** [comment]
update needed:

```
/u/qiaolei/devel/NSolver/NSolver/source/solver/initialize.cpp: In member function ‘void NSFEMSolver::NSolver<dim>::initialize() [with int dim = 2]’:
/u/qiaolei/devel/NSolver/NSolver/source/solver/NSolver.inst:1:20:   instantiated from here
/u/qiaolei/devel/NSolver/NSolver/source/solver/initialize.cpp:31:83: error: no matching function for call to ‘velocityPotential::LinearVelocityPotential<2>::LinearVelocityPotential(const dealii::SmartPointer<const dealii::parallel::distributed::Triangulation<2, 2>, void>&, const dealii::SmartPointer<const NSFEMSolver::Parameters::AllParameters<2>, void>&, ompi_communicator_t*&)’
/u/qiaolei/devel/NSolver/NSolver/source/solver/initialize.cpp:31:83: note: candidate is:
/u/qiaolei/devel/NSolver/NSolver/header/NSolver/velocityPotential/linearVelocityPotential.h:74:5: note: velocityPotential::LinearVelocityPotential<dim>::LinearVelocityPotential(dealii::SmartPointer<const dealii::parallel::distributed::Triangulation<dim> >, dealii::SmartPointer<NSFEMSolver::Parameters::AllParameters<dim> >, MPI_Comm) [with int dim = 2, MPI_Comm = ompi_communicator_t*]
/u/qiaolei/devel/NSolver/NSolver/header/NSolver/velocityPotential/linearVelocityPotential.h:74:5: note:   no known conversion for argument 2 from ‘const dealii::SmartPointer<const NSFEMSolver::Parameters::AllParameters<2>, void>’ to ‘dealii::SmartPointer<NSFEMSolver::Parameters::AllParameters<2>, void>’
/u/qiaolei/devel/NSolver/NSolver/source/solver/NSolver.inst:1:20:   instantiated from here
/u/qiaolei/devel/NSolver/NSolver/source/solver/initialize.cpp:44:81: error: no matching function for call to ‘velocityPotential::FullVelocityPotential<2>::FullVelocityPotential(const dealii::SmartPointer<const dealii::parallel::distributed::Triangulation<2, 2>, void>&, const dealii::SmartPointer<const NSFEMSolver::Parameters::AllParameters<2>, void>&, ompi_communicator_t*&)’
/u/qiaolei/devel/NSolver/NSolver/source/solver/initialize.cpp:44:81: note: candidate is:
/u/qiaolei/devel/NSolver/NSolver/header/NSolver/velocityPotential/fullVelocityPotential.h:75:5: note: velocityPotential::FullVelocityPotential<dim>::FullVelocityPotential(dealii::SmartPointer<const dealii::parallel::distributed::Triangulation<dim> >, dealii::SmartPointer<NSFEMSolver::Parameters::AllParameters<dim> >, MPI_Comm) [with int dim = 2, MPI_Comm = ompi_communicator_t*]
/u/qiaolei/devel/NSolver/NSolver/header/NSolver/velocityPotential/fullVelocityPotential.h:75:5: note:   no known conversion for argument 2 from ‘const dealii::SmartPointer<const NSFEMSolver::Parameters::AllParameters<2>, void>’ to ‘dealii::SmartPointer<NSFEMSolver::Parameters::AllParameters<2>, void>’
/u/qiaolei/devel/NSolver/NSolver/source/solver/run.cpp: In member function ‘void NSFEMSolver::NSolver<dim>::run() [with int dim = 2]’:
/u/qiaolei/devel/NSolver/NSolver/source/solver/NSolver.inst:1:20:   instantiated from here
/u/qiaolei/devel/NSolver/NSolver/source/solver/run.cpp:123:9: error: assignment of member ‘NSFEMSolver::Parameters::Refinement<2>::max_cells’ in read-only object
/u/qiaolei/devel/NSolver/NSolver/source/solver/NSolver.inst:1:20:   instantiated from here
/u/qiaolei/devel/NSolver/NSolver/source/solver/run.cpp:124:9: error: assignment of member ‘NSFEMSolver::Parameters::Refinement<2>::max_cells’ in read-only object
/u/qiaolei/devel/NSolver/NSolver/source/solver/run.cpp:128:9: error: assignment of member ‘NSFEMSolver::Parameters::Refinement<2>::max_cells’ in read-only object
/u/qiaolei/devel/NSolver/NSolver/source/solver/NSolver.inst:1:20:   instantiated from here
/u/qiaolei/devel/NSolver/NSolver/source/solver/run.cpp:617:17: error: assignment of member ‘NSFEMSolver::Parameters::Flux::numerical_flux_type’ in read-only object
make[2]: *** [source/solver/CMakeFiles/solver.dir/initialize.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [source/solver/CMakeFiles/solver.dir/run.cpp.o] Error 1
/u/qiaolei/devel/NSolver/NSolver/source/solver/refine_grid.cpp: In member function ‘void NSFEMSolver::NSolver<dim>::refine_grid() [with int dim = 2]’:
/u/qiaolei/devel/NSolver/NSolver/source/solver/NSolver.inst:1:20:   instantiated from here
/u/qiaolei/devel/NSolver/NSolver/source/solver/refine_grid.cpp:53:9: error: no matching function for call to ‘refine_and_coarsen_fixed_number(dealii::parallel::distributed::Triangulation<2, 2>&, dealii::Vector<float>&, const dealii::SmartPointer<const NSFEMSolver::Parameters::AllParameters<2>, void>&)’
/u/qiaolei/devel/NSolver/NSolver/source/solver/refine_grid.cpp:53:9: note: candidate is:
/u/qiaolei/devel/NSolver/NSolver/header/NSolver/Tools.h:115:5: note: template<int dim, class Vector> void NSFEMSolver::Tools::refine_and_coarsen_fixed_number(dealii::parallel::distributed::Triangulation<dim>&, const Vector&, const dealii::SmartPointer<NSFEMSolver::Parameters::AllParameters<dim> >&)
make[2]: *** [source/solver/CMakeFiles/solver.dir/refine_grid.cpp.o] Error 1
make[1]: *** [source/solver/CMakeFiles/solver.dir/all] Error 2
```

**Comment by QiaoLei-88** [comment]
closed via 8712bd6b699e6692721825d1a8a59f4b11a10634
